osquery/tables/system/darwin/fde_status.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright (c) 2015, Sharvil Shah
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/sql.h>
+#include <iostream>
+
+#include "osquery/tables/system/darwin/iokit_utils.h"
+
+namespace osquery {
+namespace tables {
+
+// AES-XTS is the only block algorithm supported by FileVault2
+// https://opensource.apple.com/source/xnu/xnu-2782.1.97/libkern/crypto/corecrypto_aesxts.c
+const std::string kEncryptionType = "AES-XTS";
+const std::string kDeviceNamePrefix = "/dev/";
+
+// kCoreStorageIsEncryptedKey is not publicly defined
+// or documented because CoreStorage is a private framework
+#define kCoreStorageIsEncryptedKey_ "CoreStorage Encrypted"
+
+void genFDEStatusForBSDName(const std::string& bsd_name,
+                            const std::string& uuid,
+                            QueryData& results) {
+
+  auto matching_dict =
+      IOBSDNameMatching(kIOMasterPortDefault, kNilOptions, bsd_name.c_str());
+  if (matching_dict == nullptr) {
+    CFRelease(matching_dict);
+    return;
+  }
+
+  auto service =
+      IOServiceGetMatchingService(kIOMasterPortDefault, matching_dict);
+  if (!service) {
+    IOObjectRelease(service);
+    return;
+  }
+
+  auto property =
+      IORegistryEntryCreateCFProperty(service,
```

Comment:
You can use our wrapper: [`getIOKitProperty`](https://github.com/facebook/osquery/blob/master/osquery/tables/system/darwin/iokit_utils.h#L19) if you're not bothered by a string comparison.